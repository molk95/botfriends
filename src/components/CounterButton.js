import React, { Component } from "react";

export default class CounterButton extends Component {
  state = {
    count: 0,
  };

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }
  render() {
    return <div>
        <button
        id='counter'
        color={this.props.color}
        onClick={
            () => this.setState(state => ({
                count:state.count + 1
            }))
        }> Count: {this.state.count} </button>
    </div>;
  }
}
