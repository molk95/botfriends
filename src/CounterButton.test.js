import { shallow } from "enzyme";
import React from "react";
import CounterButton from "./components/CounterButton";

describe("CounterCount", () => {
  const mockerColor = 'red';
  it("expect to render Counter Button component", () => {
    expect(shallow(<CounterButton color={mockerColor} />)).toMatchSnapshot();
  });
  it("correctly increments the counter", () => {
    const wrapper = shallow(<CounterButton color={mockerColor} />);

    wrapper.find('[id="counter"]').simulate("click");
    expect(wrapper.state()).toEqual({ count: 1 });
    wrapper.find('[id="counter"]').simulate("click");
    expect(wrapper.state()).toEqual({ count: 2 });
    expect(wrapper.props().color).toEqual('red');
  });
});
