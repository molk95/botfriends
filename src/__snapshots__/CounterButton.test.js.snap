// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterCount expect to render Counter Button component 1`] = `ShallowWrapper {}`;
